sample-apps/discount-functions-sample-app/extensions/order-discount/src/api.rs

```diff
@@ -15,102 +13,48 @@ pub mod input {
     #[derive(Clone, Debug, Deserialize)]
     #[serde(rename_all(deserialize = "camelCase"))]
     pub struct Input {
-        pub customer: Option<Customer>,
-        pub delivery_lines: Option<Vec<DeliveryLine>>,
-        pub locale: Option<String>,
-        pub merchandise_lines: Option<Vec<MerchandiseLine>>,
-        pub discount_node: Option<DiscountNode>,
+        pub discount_node: DiscountNode,
     }
 
     #[derive(Clone, Debug, Deserialize, Default)]
     pub struct DiscountNode {
-        pub metafield: Option<Metafield>,
+        pub metafield: Metafield,
     }
 
     #[derive(Clone, Debug, Deserialize, Default)]
     #[serde(rename_all(deserialize = "camelCase"))]
     pub struct Metafield {
-        pub value: Option<String>,
+        pub value: String,
     }
 
-    #[derive(Clone, Debug, Deserialize)]
-    #[serde(rename_all(deserialize = "camelCase"))]
-    pub struct Customer {
-        pub accepts_marketing: Option<Boolean>,
-        pub email: Option<String>,
-        pub id: Option<ID>,
-        pub order_count: Option<Int>,
-        pub phone: Option<String>,
-        pub tags: Option<Vec<String>>,
-        pub total_spent: Option<Money>,
-    }
-
-    #[derive(Clone, Debug, Deserialize)]
-    pub struct Money {
-        pub currency: String,
-        pub subunits: UnsignedInt64,
+    #[derive(Clone, Debug, Serialize, Deserialize)]
+    #[serde(rename_all = "camelCase")]
+    pub struct Configuration {
+        pub value: f64,
+        pub excluded_variant_ids: Vec<ID>,
     }
 
-    #[derive(Clone, Debug, Deserialize)]
-    pub struct DeliveryLine {
-        pub destination: Option<Address>,
-        pub id: Option<ID>,
-        pub subscription: Option<Boolean>,
-    }
+    impl Configuration {
+        pub const DEFAULT_VALUE: f64 = 50.0;
 
-    #[derive(Clone, Debug, Deserialize)]
-    #[serde(rename_all(deserialize = "camelCase"))]
-    pub struct Address {
-        pub city: Option<String>,
-        pub country_code: Option<String>,
-        pub phone: Option<String>,
-        pub po_box: Option<Boolean>,
-        pub province_code: Option<String>,
-        pub zip: Option<String>,
+        fn from_str(str: &str) -> Self {
+            serde_json::from_str(str).unwrap_or_default()
```

Comment:
I know someone (perhaps @nickwesselman) commented that the `unwrap_or_default()` here makes it difficult to debug and instead we should `unwrap()` (perhaps with a message)

Reply:
In this case .expect("<message">) would be great. Its like unwrap but with a message.